gems/aws-sdk-core/lib/aws-sdk-core/plugins/regional_endpoint.rb

```diff
@@ -117,6 +138,32 @@ def resolve_use_fips_endpoint(cfg)
           value ||= Aws.shared_config.use_fips_endpoint(profile: cfg.profile)
           Aws::Util.str_2_bool(value) || false
         end
+
+        # get a custom configured endpoint from ENV or configuration
+        def resolve_custom_config_endpoint(cfg)
+          config_endpoint = nil
+          ignore_configured_endpoints = cfg.ignore_configured_endpoint_urls
+          if ignore_configured_endpoints.nil?
+            ignore_configured_endpoints = ENV['AWS_IGNORE_CONFIGURED_ENDPOINT_URLS']
+            ignore_configured_endpoints ||= Aws.shared_config.ignore_configured_endpoint_urls(profile: cfg.profile)
+            ignore_configured_endpoints = Aws::Util.str_2_bool(ignore_configured_endpoints&.downcase) || false
+          end
+          unless ignore_configured_endpoints
+            service_id = cfg.api.metadata['serviceId'] || cfg.api.metadata['endpointPrefix']
+            env_service_id = service_id.gsub(" ", "_").upcase
+            if (config_endpoint = ENV["AWS_ENDPOINT_URL_#{env_service_id}"])
+              cfg.logger&.debug(
```

Comment:
This logging logic is kind of an eye sore .. is this a requirement or just a nice to have? Maybe each of these can be their own method too